crypto-ffi/bindings/jvm/src/test/kotlin/com/wire/crypto/MLSTest.kt

```diff
@@ -107,6 +105,32 @@ class MLSTest {
         cc.transaction { ctx -> ctx.createConversation(id) }
     }
 
+    @Test
+    fun parallel_transactions_are_performed_serially() = runTest() {
+        withContext(Dispatchers.Default) {
+            val (alice) = newClients(aliceId)
+            val jobs: MutableList<Job> = mutableListOf()
+            val token = "t"
+            val transactionCount = 3
+            repeat(transactionCount) {
+                jobs += launch {
+                    alice.transaction { ctx ->
+                        delay(100.milliseconds)
+                        val data = ctx.getData()?.decodeToString()?.plus(token) ?: token
+                        ctx.setData(data.toByteArray())
+                    }
+                }
```

Comment:
I have very little Kotlin experience, but I don't understand how this test distinguishes between i.e. `Mutex::lock` behavior and `RwLock::read` behavior.

Reply:
I don't understand the question. Can you expand a bit on what you mean by distinguishing between these two locks in the test?